### PR TITLE
Use annotated file A for spatial and temporal metadata instead of HDF5, drop the SAT extension. 

### PR DESCRIPTION
**Pull request by emileten**
Related Issue(s):
https://github.com/stactools-packages/nisar-sim/issues/3

Description:

- Use annotated file A for datetimes instead of HDF5 metadata
- Use annotated file A for geometries instead of HDF5 metadata
- Make the SAT extension optional, by default not using it. 

PR checklist:

- [x] Code is formatted (run scripts/format).
- [x] Code lints properly (run scripts/lint).
- [x] Tests pass (run scripts/test).
- [ ] Documentation has been updated to reflect changes, if applicable.
- [x] Examples have been updated to reflect changes, if applicable
- [x] Changes are added to the [CHANGELOG](https://github.com/stactools-packages/nisar-sim/CHANGELOG.md).

### CODE REVIEW

File: examples/winnip_31604_12061_004_120717_L090_CX_07/winnip_31604_12061_004_120717_L090_CX_07.json [hunk 1]
```diff
@@ -4,54 +4,40 @@
...
-    "sat:absolute_orbit": 12061,
-    "datetime": "2012-07-17T14:39:45.732157Z"
+    "datetime": "2012-07-17T14:36:47Z"
   },
   "geometry": {
     "type": "Polygon",
     "coordinates": [
       [
         [
-          -97.683,
```
Comment (emileten) [inline examples/winnip_31604_12061_004_120717_L090_CX_07/winnip_31604_12061_004_120717_L090_CX_07.json:15]: @jjfrench you were commenting about the order. Does it matter, given it's a polygon ? 

The values are slightly different (but we also expect that, we saw that the values were slightly different in the annotated files compared to the HDF5 metadata).

File: examples/winnip_31604_12061_004_120717_L090_CX_07/winnip_31604_12061_004_120717_L090_CX_07.json [hunk 2]
```diff
@@ -4,54 +4,40 @@
...
-          49.486
+          -98.6699,
+          49.846
         ]
       ]
     ]
   },
   "links": [
     {
-      "rel": "root",
```
Comment (emileten) [inline examples/winnip_31604_12061_004_120717_L090_CX_07/winnip_31604_12061_004_120717_L090_CX_07.json]: did you have to write these down manually @jjfrench ?

Comment (jjfrench) [reply examples/winnip_31604_12061_004_120717_L090_CX_07/winnip_31604_12061_004_120717_L090_CX_07.json]: These should get generated when running the examples script

File: examples/winnip_31604_12061_004_120717_L090_CX_07/winnip_31604_12061_004_120717_L090_CX_07.json (hunk 1 above)
Comment (jjfrench) [reply examples/winnip_31604_12061_004_120717_L090_CX_07/winnip_31604_12061_004_120717_L090_CX_07.json:15]: It matters for some applications, typically polygons should follow the "right-hand" rule

Comment (emileten) [reply examples/winnip_31604_12061_004_120717_L090_CX_07/winnip_31604_12061_004_120717_L090_CX_07.json:15]: I realize that it's obvious not _any_ order works, sorry 😅.


I am finding contradictory answers as to what the 'right hand rule' means here : clockwise or counter clockwise ? 

The GeoJSON spec seems to be saying counterclockwise for our case, so I switched to that : https://datatracker.ietf.org/doc/html/rfc7946#section-3.1.6.

File: examples/winnip_31604_12061_004_120717_L090_CX_07/winnip_31604_12061_004_120717_L090_CX_07.json (hunk 2 above)
Comment (emileten) [reply examples/winnip_31604_12061_004_120717_L090_CX_07/winnip_31604_12061_004_120717_L090_CX_07.json]: Which I had not done 👍 thanks

**Comment by emileten** [comment]
Addressed comments @jjfrench. Anything else ?

File: examples/winnip_31604_12061_004_120717_L090_CX_07/winnip_31604_12061_004_120717_L090_CX_07.json (hunk 1 above)
Comment (jjfrench) [reply examples/winnip_31604_12061_004_120717_L090_CX_07/winnip_31604_12061_004_120717_L090_CX_07.json:15]: Counterclockwise should be correct, I think that the updated geometry looks good.

File: src/stactools/nisar_sim/stac.py [hunk 3]
```diff
@@ -59,7 +59,9 @@ def create_collection() -> Collection:
     return collection
 
 
-def create_item(product_path: str, dither: str, nmode: str) -> Item:
+def create_item(
+    product_path: str, dither: str, nmode: str, sat_extension: bool = False
```
Comment (jjfrench) [inline src/stactools/nisar_sim/stac.py:63]: Just noting that sat_extension isn't doing anything here

**Review by jjfrench — APPROVED** [review]

**Comment by emileten** [comment]
I think we won't need this, as we found this solution https://github.com/stactools-packages/nisar-sim/pull/6 to https://github.com/stactools-packages/nisar-sim/issues/3

**Pull request closed without merging**